task: unwind skip_very_good_optimization tags and re-enable skipped tests

Parent: #2899
Related: #3081 (introduces the tag set into main CI)
Part of: #2890 CI speed umbrella

## Scope

When #3081 lands, ~55 test files will ship tagged with `@Tags(['skip_very_good_optimization'])`. These files opt out of Very Good CLI's merged-isolate optimizer because they have cross-file state contamination (singletons, platform channels, `ImageCache`, `FlutterError.onError`, `IntegrationTestWidgetsFlutterBinding`, or shared mock/provider setup). Each tagged file still runs — it just runs in its own isolate, which costs a full Dart VM startup per file and is what forced `--concurrency=2` in the runner.

This issue tracks the incremental work to:

1. Fix the underlying isolation problem for each file and remove the tag.
2. Re-enable two tests that were marked `skip: true` as part of the VGV migration.

Each file can be handled independently in its own PR. The goal is to drive the tag count toward 0 over time so the optimizer can merge more files into the shared isolate and the `--concurrency` ceiling can be raised.

## Why the tag exists

VGV's `very_good test --optimization` concatenates eligible test files into a single `_test.dart` entry point and runs them in one Dart isolate. Files that depend on process-wide state that leaks between tests (registered platform channels that aren't cleared, static singletons without `resetInstance()`, `PaintingBinding.instance.imageCache` pollution, drift AppDatabase locks, path_provider plugin registration) fail when run this way. Tagging with `skip_very_good_optimization` tells VGV to keep them isolated.

Tagging is a valid escape hatch — it is not a bug — but the debt is worth paying down because every tagged file adds ~5–10s of Dart VM startup to CI.

## Unwind buckets

Work is grouped by the most likely root cause. Pick a bucket, fix the tests in it, remove the tag, and open a PR per bucket (or smaller).

### A. Integration-style tests (prefer relocation over unwind)

These files are tagged with both `integration` and `skip_very_good_optimization`. Most look like they belong under `mobile/integration_test/` rather than `mobile/test/`. First step is to decide per file: move to `integration_test/` (and delete from `test/`) or rewrite as a unit/widget test.

- [ ] `mobile/test/models/thumbnail_url_preservation_test.dart`
- [ ] `mobile/test/models/upload_publish_race_condition_test.dart`
- [ ] `mobile/test/models/video_publish_flow_test.dart`
- [ ] `mobile/test/providers/video_playback_stop_integration_test.dart`
- [ ] `mobile/test/repositories/profile_cache_sync_test.dart`
- [ ] `mobile/test/router/profile_me_redirect_integration_test.dart`
- [ ] `mobile/test/screens/hashtag_grid_view_simple_test.dart`
- [ ] `mobile/test/screens/profile_menu_drafts_navigation_test.dart`
- [ ] `mobile/test/services/account_deletion_flow_test.dart`
- [ ] `mobile/test/services/auto_draft_complete_flow_test.dart`
- [ ] `mobile/test/services/blossom_resumable_upload_integration_test.dart`
- [ ] `mobile/test/services/bug_report_worker_api_test.dart`
- [ ] `mobile/test/services/cache_first_query_test.dart`
- [ ] `mobile/test/services/hashtag_filtering_integration_test.dart`
- [ ] `mobile/test/services/reactive_pagination_test.dart`
- [ ] `mobile/test/services/video_event_service_with_event_router_test.dart`
- [ ] `mobile/test/services/video_event_thumbnail_integration_test.dart`
- [ ] `mobile/test/widgets/ios_network_image_deadlock_test.dart`

### B. Singleton / static-state leaks

Services or notifiers that hold process-wide state and need a `@visibleForTesting resetInstance()` in production code plus `tearDown` in the test. See PR #3082 for the established pattern.

- [ ] `mobile/test/services/notification_service_enhanced_test.dart`
- [ ] `mobile/test/services/screen_analytics_service_test.dart`
- [ ] `mobile/test/services/feed_performance_tracker_stale_session_test.dart`
- [ ] `mobile/test/services/background_activity_manager_test.dart`
- [ ] `mobile/test/services/draft_storage_service_test.dart`
- [ ] `mobile/test/services/web_auth_service_bunker_test.dart`
- [ ] `mobile/test/services/notification_persistence_test.dart`
- [ ] `mobile/test/services/clip_library_service_test.dart`
- [ ] `mobile/test/services/audio_extraction_service_test.dart`
- [ ] `mobile/test/services/video_event_service_deduplication_test.dart`
- [ ] `mobile/test/unit/services/video_event_service_subscription_test.dart`
- [ ] `mobile/test/unit/services/upload_manager_get_by_path_test.dart`
- [ ] `mobile/test/unit/services/bug_report_service_upload_logs_test.dart`
- [ ] `mobile/test/services/upload_initialization_helper_test.dart`
- [ ] `mobile/test/services/upload_manager_sandbox_test.dart`

### C. Platform channel / plugin state

Files that register platform-channel mocks at file scope instead of per-test. Fix is to move the `setMockMethodCallHandler` call into `setUp` and clear it in `tearDown`.

- [ ] `mobile/test/services/video_thumbnail_service_test.dart`
- [ ] `mobile/test/services/video_editor/video_editor_split_service_test.dart`
- [ ] `mobile/test/infrastructure/drift_setup_test.dart`
- [ ] `mobile/test/infrastructure/schema_test.dart`
- [ ] `mobile/test/providers/clip_manager_provider_test.dart`
- [ ] `mobile/test/providers/video_recorder_provider_test.dart`
- [ ] `mobile/test/providers/vine_recording_provider_auto_draft_test.dart`
- [ ] `mobile/test/screens/video_recorder_screen_test.dart`

### D. Widget tests with shared mock / provider / fake state

Profile grids, search results, and feed widgets that share fakes at file scope or leak `ProviderContainer` state between tests.

- [ ] `mobile/test/widgets/profile/profile_collabs_grid_test.dart`
- [ ] `mobile/test/widgets/profile/profile_comments_grid_test.dart`
- [ ] `mobile/test/widgets/profile/profile_liked_grid_test.dart`
- [ ] `mobile/test/widgets/profile/profile_reposts_grid_test.dart`
- [ ] `mobile/test/widgets/profile_drafts_button_test.dart`
- [ ] `mobile/test/widgets/share_video_menu_comprehensive_test.dart`
- [ ] `mobile/test/widgets/video_feed_item/actions/share_action_button_test.dart`
- [ ] `mobile/test/widgets/video_feed_item/video_overlay_actions_description_test.dart`
- [ ] `mobile/test/widgets/video_metadata/video_metadata_tags_input_test.dart`
- [ ] `mobile/test/widgets/video_metadata/video_metadata_user_chip_test.dart`
- [ ] `mobile/test/widgets/global_error_widget_test.dart`
- [ ] `mobile/test/screens/discover_lists_screen_test.dart`
- [ ] `mobile/test/screens/feed/video_feed_page_test.dart`
- [ ] `mobile/test/screens/feed/pooled_fullscreen_video_feed_screen_test.dart`
- [ ] `mobile/test/screens/search_results/view/search_results_page_test.dart`
- [ ] `mobile/test/screens/search_results/widgets/videos_section_test.dart`
- [ ] `mobile/test/screens/explore_screen_hashtag_loading_test.dart`
- [ ] `mobile/test/notifications/widgets/notification_avatar_stack_test.dart`
- [ ] `mobile/test/notifications/widgets/notification_list_item_test.dart`

### E. Re-enable skipped tests

Two tests were marked `skip: true` during the VGV migration and need individual attention before being re-enabled.

- [ ] `mobile/test/services/seed_media_preload_service_test.dart` — the "copies-bundled-videos" test is skipped. Comment in code: *"Re-enable after the asset mock is moved into `setUpAll` or a dedicated `AssetBundle` override."*
- [ ] `mobile/test/screens/sounds_screen_test.dart` — the "navigating to detail stops current preview" test is skipped. Comment in code: *"Move shared stubs into this group's `setUp` or an `AudioPlaybackService` fake."*

## Implementation Notes

- Prefer **one PR per bucket**, or per small sub-group inside a bucket, so reviews stay small.
- Use PR #3082 as the reference pattern for singleton fixes (`@visibleForTesting resetInstance()` + `setUp`/`tearDown`).
- After removing a tag, verify the file passes under `very_good test --optimization --concurrency=2 --exclude-tags integration --test-randomize-ordering-seed random` run multiple times locally to exercise different orderings.
- If a file genuinely cannot be un-isolated (e.g. it uses `IntegrationTestWidgetsFlutterBinding`), leave the tag and add a code comment explaining why it is permanent.

## Acceptance Criteria

- [ ] Bucket A files either relocated to `mobile/integration_test/` or rewritten to not depend on real I/O
- [ ] Bucket B/C/D files have their underlying isolation issue fixed and the tag removed
- [ ] Bucket E tests re-enabled (both `skip: true` markers removed)
- [ ] Final `skip_very_good_optimization` tag count is documented (target: files that legitimately cannot be un-isolated, with inline comment each)
- [ ] `mobile_ci.yaml` `--concurrency` raised above 2 once tag count is low enough

## Sequencing

Not blocking anything. Work can proceed in any order, in parallel with other backlog. This is pure CI-speed debt paydown.

## Correction to the Bucket A strategy in #3280

Attempted the "Bucket A strike" — stripping `skip_very_good_optimization` from the 17 files dual-tagged with `integration`, on the premise that `--exclude-tags integration` was already excluding them from the VGV test job.

**That premise is wrong.** Empirical evidence from a local VGV merged-suite run (seed 42) on a branch with the 17 tags stripped:

```
Failing Tests:
 - test/widgets/ios_network_image_deadlock_test.dart
   - [ERROR] iOS Network Image Deadlock Prevention Multiple concurrent network images should load without hanging (iOS regression test) (×7)
 ...
02:21 +7593 -11 ~401: Some tests failed.
```

The `ios_network_image_deadlock_test.dart` tests **ran** and failed in the merged isolate despite the file having `@Tags(['integration'])` at line 1 (library-level).

### Why the exclusion doesn't work under VGV optimization

`very_good test --optimization` generates a meta-`_test.dart` that imports all eligible files into a single compilation unit. The library-level `@Tags` annotation on each original file is a **library annotation** — it does not propagate to individual tests when those tests are hoisted into the meta-file's scope. VGV's `--exclude-tags` flag filters at test-runtime based on test-level tags, not library-level.

The `skip_very_good_optimization` tag is different: it's recognized by the **optimizer at merge time** (before any code runs) to keep files out of the meta-file entirely. That's what actually keeps the Bucket A files out of the CI test job today — not the `integration` tag.

Consequence: the Bucket A strategy from #3280 — "strike the redundant `skip_very_good_optimization`, keep only `integration`" — doesn't work as a pure hygiene PR. Each of the 17 files needs either a proper fix for the merged isolate (same treatment as buckets B/C/D) OR the `integration` tag moved from library-level to per-test (`testWidgets('...', tags: ['integration'], ...)`).

Reverted the attempted strike PR locally; main remains at 61 tagged files. Will update the brainstorm doc (#3280) in a follow-up with this correction so the remaining work is planned against the right premise.

### Implication for the rest of the unwind

Tag-count projection was previously:
- 61 → 54 (after 7 individual untag PRs) → 37 (after Bucket A strike) → ~37 remaining.

Corrected projection:
- 61 → 54 (after 7 individual untag PRs) → Bucket A files need individual investigation, similar to other buckets.

No change to the already-shipped PRs (#3282, #3284, #3287, #3288, #3289, #3290, #3291, #3292). They stand.

--- a/mobile/test/models/upload_publish_race_condition_test.dart
+++ b/mobile/test/models/upload_publish_race_condition_test.dart
@@ -1,4 +1,3 @@
-@Tags(['skip_very_good_optimization', 'integration'])
 // ABOUTME: Test verifying race condition is fixed - upload completes before publishing
 // ABOUTME: Ensures videoId and cdnUrl are populated when publishDirectUpload is called
 import 'package:flutter_test/flutter_test.dart';

